src/jobs/processClinicalEvent.ts

```diff
@@ -17,6 +23,24 @@ const clinicalUpdateEvent = async (clinicalEvent: ClinicalUpdateEvent) => {
         0} donors`,
     );
 
+    const indexer = await getIndexer();
+
+    // get files for donors
+    const files = await fileService.getFiles({ include: { donors: donorIds } });
+    // we only need to do state updates for restricted files
+    const unreleasedFiles = files.filter(isUnreleased);
+
+    // for each file check if they should be released and then reindex
```

Comment:
Agree with @mistryrn's comment. this step by step comment documentation is pretty good

Reply:
here i say restricted and then filtered by unreleased... the code is caught between two ideas:

right now, we dont index clinical data, we just index once we get core complete, so filtering by unreleased is good but this will change to `isRestricted` once we do index clinical data... not sure if i should change this or not, but i'll clean up the comments to reflect this.

Reply:
IMO These are pretty good, even if they don't paint that whole picture.
Maybe I'd add a TODO explaining that ☝️ is coming up, which may help document the changes later.